analysis/topeft_run2/analysis_processor.py

```diff
@@ -536,12 +537,16 @@ def process(self, events):
             selections.add("2los_CRZ", (events.is2l_nozeeveto & charge2l_0 & sfosz_2l_mask & bmask_exactly0med & pass_trg))
 
             # 3l selection
-            selections.add("3l_p_offZ_1b", (events.is3l & charge3l_p & ~sfosz_3l_mask & bmask_exactly1med & pass_trg))
-            selections.add("3l_m_offZ_1b", (events.is3l & charge3l_m & ~sfosz_3l_mask & bmask_exactly1med & pass_trg))
-            selections.add("3l_p_offZ_2b", (events.is3l & charge3l_p & ~sfosz_3l_mask & bmask_atleast2med & pass_trg))
-            selections.add("3l_m_offZ_2b", (events.is3l & charge3l_m & ~sfosz_3l_mask & bmask_atleast2med & pass_trg))
-            selections.add("3l_onZ_1b", (events.is3l & sfosz_3l_mask & bmask_exactly1med & pass_trg))
-            selections.add("3l_onZ_2b", (events.is3l & sfosz_3l_mask & bmask_atleast2med & pass_trg))
+            selections.add("3l_p_offZ_low_1b", (events.is3l & charge3l_p & ~sfosz_3l_OnZ_mask & sfosz_3l_OffZ_low_mask & bmask_exactly1med & pass_trg))
+            selections.add("3l_p_offZ_high_1b", (events.is3l & charge3l_p & ~sfosz_3l_OnZ_mask & ~sfosz_3l_OffZ_low_mask & bmask_exactly1med & pass_trg))
+            selections.add("3l_m_offZ_low_1b", (events.is3l & charge3l_m & ~sfosz_3l_OnZ_mask & sfosz_3l_OffZ_low_mask & bmask_exactly1med & pass_trg))
+            selections.add("3l_m_offZ_high_1b", (events.is3l & charge3l_m & ~sfosz_3l_OnZ_mask & ~sfosz_3l_OffZ_low_mask & bmask_exactly1med & pass_trg))
+            selections.add("3l_p_offZ_low_2b", (events.is3l & charge3l_p & ~sfosz_3l_OnZ_mask & sfosz_3l_OffZ_low_mask & bmask_atleast2med & pass_trg))
+            selections.add("3l_p_offZ_high_2b", (events.is3l & charge3l_p & ~sfosz_3l_OnZ_mask & ~sfosz_3l_OffZ_low_mask & bmask_exactly2med & pass_trg))
+            selections.add("3l_m_offZ_low_2b", (events.is3l & charge3l_m & ~sfosz_3l_OnZ_mask & sfosz_3l_OffZ_low_mask & bmask_atleast2med & pass_trg))
+            selections.add("3l_m_offZ_high_2b", (events.is3l & charge3l_m & ~sfosz_3l_OnZ_mask & ~sfosz_3l_OffZ_low_mask & bmask_exactly2med & pass_trg))
+            selections.add("3l_onZ_1b", (events.is3l & sfosz_3l_OnZ_mask & bmask_exactly1med & pass_trg))
+            selections.add("3l_onZ_2b", (events.is3l & sfosz_3l_OnZ_mask & bmask_atleast2med & pass_trg))
```

Comment:
I liked @kmohrman's comment that we could keep the original selections. We could either
1. Restore them here
2. Do some merging the datacard step